lib/stack_master/parameter_resolvers/ami_finder.rb

```diff
@@ -0,0 +1,36 @@
+module StackMaster
+  module ParameterResolvers
+    class AmiFinder
+      def initialize(region)
+        @region = region
+        @owner = ['self']
+      end
+
+      def build_filters(value, prefix = nil)
+        owner_id = []
+        filters = value.split(',').map do |name_with_value|
+          name, value = name_with_value.strip.split('=')
+          owner_id.push value if name == 'owner_id'
+          name = prefix ? "#{prefix}:#{name}" : name
+          { name: name, values: [value] }
+        end
+        @owner = owner_id unless owner_id.empty?
+        filters
+      end
```

Comment:
> I think the owners option (not the filter) sets the scope of what the filters can ... filter

Correct

> How would you select an image owned by amazon or in the aws marketplace with this resolver?

The only way at the moment would be to use the account ID.

I can't think of a better way to do it and since it's not possible at the moment I still think we've made an improvement.